comment/src/lib.rs

```diff
@@ -0,0 +1,124 @@
+#![allow(clippy::or_fun_call)]
+use std::ffi::OsString;
+use std::str::FromStr;
+
+use anyhow::anyhow;
+
+use radicle_common::args::{Args, Error, Help};
+use radicle_common::{
+    cobs::{self, issue::*, CobIdentifier, CommentId, Store as _},
+    keys, person, profile, project,
+};
+use radicle_terminal as term;
+
+pub const HELP: Help = Help {
+    name: "comment",
+    description: env!("CARGO_PKG_DESCRIPTION"),
+    version: env!("CARGO_PKG_VERSION"),
+    usage: r#"
+Usage
+
+    rad comment <id> [--description <text>] [--reply-to <index>]
+
+Options
+
+    --description <text>    Comment text
+    --reply-to <index>      Index of comment writing a reply for
+    --help                  Print help
+"#,
+};
+
+#[derive(Debug)]
+pub struct Options {
+    pub id: CobIdentifier,
+    pub description: Option<String>,
+    pub reply_index: Option<CommentId>,
+}
+
+impl Args for Options {
+    fn from_args(args: Vec<OsString>) -> anyhow::Result<(Self, Vec<OsString>)> {
+        use lexopt::prelude::*;
+
+        let mut parser = lexopt::Parser::from_args(args);
+        let mut id: Option<CobIdentifier> = None;
+        let mut description: Option<String> = None;
+        let mut reply_index: Option<CommentId> = None;
+
+        while let Some(arg) = parser.next()? {
+            match arg {
+                Long("help") => {
+                    return Err(Error::Help.into());
+                }
+                Long("description") => {
+                    description = Some(parser.value()?.to_string_lossy().into());
+                }
+                Long("reply-to") => {
+                    let idx = parser
+                        .value()?
+                        .parse::<usize>()
+                        .map_err(|_| anyhow!("index for `--reply-to` can't be parsed as usize"))?;
+
+                    reply_index = Some(CommentId::from(idx));
+                }
+                Value(val) if id.is_none() => {
+                    let val = val
+                        .to_str()
+                        .ok_or_else(|| anyhow!("issue id specified is not UTF-8"))?;
+
+                    id = Some(
+                        CobIdentifier::from_str(val)
+                            .map_err(|_| anyhow!("invalid issue id '{}'", val))?,
```

Comment:
we can replace "issue" with "object" everywhere here so it applies to patches too.

Reply:
https://github.com/radicle-dev/radicle-cli/compare/8f54ce1f4a372c8383153cf195678c6a8f0c5aad..43ca84d9b690109a28a25738a4a9e37db335b32b